File: src/main/java/it/intesys/codylab/rookie/commandline/person.py
from datetime import datetime
from typing import Optional


class Person:
    def __init__(
        self,
        id: int = 0,
        name: str = "",
        surname: str = "",
        registrationDate: Optional[datetime] = None,
    ) -> None:
        self.id = id
        self.name = name
        self.surname = surname
        self.registrationDate = registrationDate

    def toString(self, longFormat: bool) -> str:
        if longFormat:
            return (
                "\nPerson (id: "
                + str(self.id)
                + ",\n"
                + "name: "
                + str(self.name)
                + ",\n"
                + "surname: "
                + str(self.surname)
                + ",\n"
                + "registrationDate: "
                + str(self.registrationDate)
                + ")\n"
            )
        else:
            return "Person (id: " + str(self.id) + ")"

File: src/main/java/it/intesys/codylab/rookie/commandline/Person.java
package it.intesys.codylab.rookie.commandline;

import java.time.Instant;

public class Person {
    private static int lastId;

    int id;
    String name;
    String surname;
    Instant registrationDate;

    public Person () {
        lastId = lastId + 1;
        this.id = lastId;
        registrationDate = java.time.Instant.now();
    }

    public Person (String name, String surname) {
        this ();
        this.name = name;
        this.surname = surname;
    }

    public Person (String name, String surname, Instant registrationDate) {
        this (name, surname);
        this.registrationDate = registrationDate;
    }

    String toString(boolean longFormat) {
        if (longFormat) {
            return "\nPerson (id: " + id + ",\n" +
                "name: " + name + ",\n" +
                "surname: " + surname + ",\n" +
                "registrationDate: " + registrationDate + ")\n";
        } else {
            return "Person (id: " + id  + ")";
        }
    }

    public static void resetLastId () {
        lastId = 0;
    }

    public static void incrementLastId (int lastIdIncrement) {
        lastId += lastIdIncrement;
    }

}
